src/pcre2_jit_compile.c

```diff
@@ -1838,7 +1838,8 @@ if (*next == OP_BRAZERO || *next == OP_BRAMINZERO)
 
       if (i == max)
         {
-        common->private_data_ptrs[max_end - common->start - LINK_SIZE] = next_end - max_end;
+        /* Patterns must fit into an int32 even for link-size=4. */
```

Comment:
I think this comment I have just added is correct.

I was worried about whether there's any case where a pattern can by 3GiB big (ie, bigger than an S32 can hold, but smaller than the U32 maximum for a pattern with link-size=4).

I can see there's some nice code elsewhere:

> #define MAX_PATTERN_SIZE (1 << 30)   /* Keep it positive */

("Keep it positive" clearly means, "don't use `1<<32`" as the max size, but make sure it's comfortably smaller than that.)

This apparently ensures that you can't get in this situation, where you have a pattern whose compiled size is too large to fit in an int.